reppy/cache/__init__.py

```diff
@@ -83,9 +102,17 @@ def allowed(self, url, agent):
 
     def fetch(self, url):
         '''Return (expiration, Robots) for the robots.txt at the provided URL.'''
+        expires = [0]
+
+        def check_ttl_hook(response):
+            if not isinstance(response, Exception):
+                expires[0] = self.ttl_policy.expires(response)
+            if self.after_response_hook is not None:
+                self.after_response_hook(response)
+
         robots = Robots.fetch(
-            url, ttl_policy=self.ttl_policy, *self.args, **self.kwargs)
-        return (robots.expires, robots)
+            url, after_response_hook=check_ttl_hook, *self.args, **self.kwargs)
```

Comment:
I agree that this doesn't feel great, but I don't see any reason not to trust it. We could also have a method off of `Robots` that returns a pair of `(response, obj)`, but I think this is fine as is.